### PR TITLE
manifest: update trusted-firmware-m to latest upstream

### PR DESCRIPTION
**Pull request by ioannisg**
Synchronize TF-M to the latest ustream main branch.
This brings TF-M support for BL5340 Dev Kit in Zephyr.

The PR fixes also a minor issue in the psa_level1 sample.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

### CODE REVIEW

**Comment by ioannisg** [comment]
FYI @greg-leach

**Review by andvib — APPROVED** [review]

**Review by microbuilder — APPROVED** [review]
Tested with `mps2_an521_nonsecure` and `lpcxpresso55s69_ns`.

**Comment by microbuilder** [comment]
I don't really love not being on a fixed release for certification reasons, since it makes the certificate reference a mess, but it does pull in some other changes that are useful before 1.4.0 is tagged upstream (which will be post Zephyr 2.6.0).

**Comment by ioannisg** [comment]
> I don't really love not being on a fixed release for certification reasons, since it makes the certificate reference a mess, but it does pull in some other changes that are useful before 1.4.0 is tagged upstream (which will be post Zephyr 2.6.0).

Let's hold this off, until we know we will be able to merge the support for the BL5340 DK. Otherwise, this could be done post v2.6.0.

**Comment by ioannisg** [comment]
> Tested with `mps2_an521_nonsecure` and `lpcxpresso55s69_ns`.

Thanks. Myself, tested this on nrf5340dk, nrf9160dk and nucleo_l552ze_q_ns

**Review by erwango — APPROVED** [review]

**Review by oyvindronningstad — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Pull request merged**
